cpe_workspaceone/resources/cpe_workspaceone.rb

```diff
@@ -42,9 +42,22 @@ def uninstall?
 
   def enforce_mdm_profiles
     return unless node['cpe_workspaceone']['mdm_profiles']['enforce']
+
     macos_enforce_mdm_profiles if node.macos?
   end
 
+  def set_cli_config
+    unless node.ws1_hubcli_exists
+      Chef::Log.warn('cpe_workspaceone - hubcli path does not exist, cannot enforce MDM profiles!')
+      return
+    end
+
+    prefs = node['cpe_workspaceone']['cli_prefs'].reject { |_, v| v.nil? }
+    prefs.each do |flag, val|
+      node.hubcli_execute("config --set #{flag} #{val}")
```

Comment:
Haven't tested, but this looks like it won't be idempotent. Am I wrong?

Reply:
The current config state shouldn't change what gets set, and what gets set should be the same every time, so I think you are wrong in regards to the golden path.

```
➜  cpe-chef-cookbooks git:(master) sudo hubcli config --set menubar-icon false
menubar-icon:false
➜  cpe-chef-cookbooks git:(master) sudo hubcli config --set menubar-icon false
menubar-icon:false
```

However, in the case that the operator sets, first, `'menubar-icon' => false`, then `'menubar-icon' => nil`, the previous `false` value will remain. I'm ok with that. The only solution I see would be defining default values in case of `nil`, and I'm not sure that's reliably possible in the case of `awcm-url` and `server-url`.

Reply:
Is there a way to do a `config --get`?

Reply:
Alternatively, we could just refuse to manage those keys, since they're massive footguns. Then we could set Airwatch's defaults for the other three (`checkin-interval:60`, `menubar-icon:false`, `sample-interval:60`) in the case of nil. That's fully idempotent.

Reply:
@natewalck yes

Reply:
What about this? 8e597653d99af4f9a0a20aaea547164ab333ca62

```
[2019-12-26T16:45:07-07:00] WARN: cpe_workspaceone - refusing to manage unknown cli preference 'awcm-url'
[2019-12-26T16:45:07-07:00] WARN: cpe_workspaceone - Error: Invalid value for option --set: menubar-icon (foobar) - setting default
```

@natewalck are you suggesting checking the config before setting values, and using that as the default?